src/transaction/types/input/repr.rs

```diff
@@ -25,7 +25,7 @@ impl InputRepr {
     pub const fn owner_offset(&self) -> Option<usize> {
         match self {
             Self::Coin => Some(INPUT_COIN_OWNER_OFFSET),
-            Self::Message => None,
+            Self::Message => Some(INPUT_MESSAGE_RECIPIENT_OFFSET),
```

Comment:
This isn't really an owner but a recipient. I don't recommend we mix things because we create a magic box effect (that we don't know what exactly is happening internally in the function, provided only its name)